NPE when using Alt+F9 on the type of a field

`@Inject Foo bar;`

'Foo' was selected when pressing Alt+F9 "Select querries for Ferret".

<pre>
Exception:org.eclipse.swt.SWTException: Failed to execute runnable (java.lang.NullPointerException)
	 at org.eclipse.swt.SWT.error(SWT.java:4533)
	 at org.eclipse.swt.SWT.error(SWT.java:4448)
	 at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:185)
	 at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4211)
	 at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3827)
	 at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1121)
	 at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	 at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1022)
	 at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:150)
	 at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:693)
	 at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	 at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:610)
	 at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	 at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:138)
	 at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	 at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	 at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	 at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	 at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	 at sun.reflect.NativeMethodAccessorImpl.invoke0(null:-2)
	 at sun.reflect.NativeMethodAccessorImpl.invoke(null:-1)
	 at sun.reflect.DelegatingMethodAccessorImpl.invoke(null:-1)
	 at java.lang.reflect.Method.invoke(null:-1)
	 at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:673)
	 at org.eclipse.equinox.launcher.Main.basicRun(Main.java:610)
	 at org.eclipse.equinox.launcher.Main.run(Main.java:1519)
Caused by: java.lang.NullPointerException: null
	 at ca.ubc.cs.ferret.views.QueriesDossierView.promptForDesiredQueries(QueriesDossierView.java:1501)
	 at ca.ubc.cs.ferret.views.PopulatedAskFerretWorkbenchAction.process(PopulatedAskFerretWorkbenchAction.java:11)
	 at ca.ubc.cs.ferret.AbstractAskFerretAction.performQuery(AbstractAskFerretAction.java:50)
	 at ca.ubc.cs.ferret.AskFerretWorkbenchAction$1.run(AskFerretWorkbenchAction.java:45)
	 at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:35)
	 at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:182)
	 at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4211)
	 at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3827)
	 at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1121)
	 at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	 at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1022)
	 at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:150)
	 at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:693)
	 at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	 at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:610)
	 at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	 at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:138)
	 at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	 at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	 at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	 at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	 at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	 at sun.reflect.NativeMethodAccessorImpl.invoke0(null:-2)
	 at sun.reflect.NativeMethodAccessorImpl.invoke(null:-1)
	 at sun.reflect.DelegatingMethodAccessorImpl.invoke(null:-1)
	 at java.lang.reflect.Method.invoke(null:-1)
	 at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:673)
	 at org.eclipse.equinox.launcher.Main.basicRun(Main.java:610)
	 at org.eclipse.equinox.launcher.Main.run(Main.java:1519)
</pre>

--- a/ca.ubc.cs.ferret/src/ca/ubc/cs/ferret/views/QueriesDossierView.java
+++ b/ca.ubc.cs.ferret/src/ca/ubc/cs/ferret/views/QueriesDossierView.java
@@ -38,6 +38,7 @@ import com.google.common.base.Strings;
 import com.google.common.collect.Multimap;
 import com.google.common.collect.MultimapBuilder;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.Comparator;
@@ -198,6 +199,9 @@ public class QueriesDossierView extends ViewPart
     protected IPropertyChangeListener preferencesChangeListener;
 	protected ISphereFactory sphereFactory;
 
+	/* Safe to use even if the viewer is not yet set */
+	private NameComparator queryComparator;
+
 	class NameComparator extends ViewerComparator {
 
         @Override
@@ -235,7 +239,7 @@ public class QueriesDossierView extends ViewPart
 	    viewer.getTree().setLinesVisible(true);
 		viewer.setContentProvider(contentProvider = new DossierContentProvider());
 		viewer.setLabelProvider(labelProvider = new DossierLabelProvider());
-		viewer.setComparator(new NameComparator());
+		viewer.setComparator(queryComparator);
 		viewer.setInput(null);
 		viewer.getControl().setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
 //				new GridData(GridData.FILL_HORIZONTAL));
@@ -1268,6 +1272,8 @@ public class QueriesDossierView extends ViewPart
 	@Override
 	public void init(IViewSite site, IMemento memento) throws PartInitException {
 		super.init(site, memento);
+		queryComparator = new NameComparator();
+
 		viewStateMemento = memento;
 		try {
 			String sphereFactoryId = memento == null ? null : memento.getString("sphereFactory");
@@ -1504,7 +1510,7 @@ public class QueriesDossierView extends ViewPart
         Consultation c = getConsultancy().createConsultation(objects, getSphere());
         c.buildConceptualQueries(new NullProgressMonitor());
         IConceptualQuery cqs[] = c.getConceptualQueries();
-        viewer.getSorter().sort(viewer, cqs);
+        Arrays.sort(cqs, (a, b) -> queryComparator.compare(viewer, a, b));
 
         ILabelProvider lp = new WorkbenchAdapterLabelProvider() {
 			@Override

